doofinder/includes/api/class-store-api.php

```diff
@@ -177,10 +249,138 @@ public function get_product_datatype()
                     "type" => "wordpress",
                     "options" => [
                         "feed_type" => "product",
-                        "url" =>  get_bloginfo('url')
+                        "url" => $this->language->get_home_url($language)
+                    ]
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * Generates the post datatype structure.
+     *
+     * @return array The post datatype structure.
+     */
+    public function get_post_datatype($language)
+    {
+        return [
+            "name" => "post",
+            "preset" => "generic",
+            "datasources" => [
+                [
+                    "type" => "wordpress",
+                    "options" => [
+                        "feed_type" => "post",
+                        "url" =>  $this->language->get_home_url($language)
                     ]
                 ]
             ]
         ];
     }
+
+    private function get_callback_urls($api_keys, $primary_language)
+    {
+        $callback_urls = [];
+        $currency = 'EUR';
+        foreach ($api_keys as $item) {
+            $code = $item['lang']['locale'] ?? $item['lang']['code'] ?? $primary_language;
+            $lang = Helpers::get_language_from_locale($code);
+            $code = Helpers::format_locale_to_hyphen($code);
+            $callback_urls[$code][$currency] = self::build_callback_url(
+                $this->language->get_home_url($lang),
+                '/wp-json/doofinder/v1/index-status/?token=' . $this->api_key
+            );
+        }
+        return $callback_urls;
+    }
+
+    /**
+     * This method takes the base url and adds
+     *
+     * @param [type] $base_url
+     * @param [type] $endpoint_path
+     * @return void
+     */
+    private static function build_callback_url($base_url, $endpoint_path)
```

Comment:
What is the reason to create a private static function?

Reply:
In fact, it doesn't have too much sense, I've adjusted some methods to be private and non-static